docs/new/develop/indexes.md

```diff
@@ -0,0 +1,99 @@
+When you execute your Substreams for the first time, you are reading the data stored in the flat files of the Substreams provider. 
+
+To improve the performance, the data accessed by the Substreams is cached, so that the second or the third time that you run the Substreams, you can read from the cache, thus saving time and money. This behavior is illustrated in the following diagram.
+
+<figure><img src="../../.gitbook/assets/develop/cache-substreams-flow.png" width="100%" /></figure>
+
+This data indexing is done implicitly every time you run a Substreams for the first time, but Substreams also allows you to explicitly specify which specific data you want to index for further performance improvement.
+
+## Indexes
+
+Substreams has recently introduced the concept of _indexed modules_. An indexed module is a module that has been pre-cached for some specific data. Let's see it with an example!
+
+Consider that you want to retrieve all the Ethereum events matching a specific address. Usually, in every block, you would iterate through all the logs  looking for those where `log.address == ADDRESS`.
+
+With indexing, you could have a pre-cached module with the information of all the event addresses in the block. Instead of reading the full Ethereum block, you can search in the events index (event cache) and avoid decoding the data of those blocks that do not contain events you are interested in.
+
+In the following diagram, you can see three blocks with their corresponding data. Consider that you want to retrieve all the events where `log.address == 0xcd2...`. In a non-indexed module, you would have to decode the data of every block, iterate over the logs, and search if there is any log matching the condition.
+
+<figure><img src="../../.gitbook/assets/develop/cache-block-1.png" width="100%" /></figure>
+
+On the other hand, in an indexed module, the events of every block are pre-cached in a special store, so when you look for events where `log.address == 0xcd2...`, you can simply search in the index store of the block. If the event is contained, within the block, then you decode the data. If not, you skip it.
+
+In the following diagram, `Block 1` and `Block 2` contain an event where `log.address == 0xcd2...`, but `Block 3` does not.
+
+<figure><img src="../../.gitbook/assets/develop/cache-block-2.png" width="100%" /></figure>
+
+### Create a Custom Index
+
+Anyone can create an indexed module. All you need to do is create a Substreams that indexes the piece of data of your choice. For example, let's take a look at the `index_events` module from the [Ethereum Foundational Modules GitHub repository](https://github.com/streamingfast/substreams-foundational-modules/blob/develop/ethereum-common/substreams.yaml#L53).
+
+The flow of indexing the events of a block looks like the following:
```

Comment:
The following flow is a good, as it explains a "good way to use an index" (in conjunction with: a first module that reduces the amount of data and a 3rd module that filters the events found in the blocks matching the index)

We should insist that this is one usage pattern for index modules, it is not the only way of using an index!
again, this results from the confusion between the role of each of the 3 modules in there...